go/fn/errors.go

```diff
@@ -39,11 +39,12 @@ func (e *errKubeObjectFields) Error() string {
 
 // errSubObjectFields raises if the SubObject operation panics.
 type errSubObjectFields struct {
+	obj    *SubObject
 	fields []string
 }
 
 func (e *errSubObjectFields) Error() string {
-	return fmt.Sprintf("SubObject has unmatched field type: `%v", strings.Join(e.fields, "/"))
+	return fmt.Sprintf("SubObject has unmatched field type: `%v`, relative path to parent kubeObject(group=%v, version=%v, kind=%v) is `%v`", strings.Join(e.fields, "/"), e.obj.gvk.Group, e.obj.gvk.Version, e.obj.gvk.Kind, strings.Join(e.obj.nodePath, "/"))
```

Comment:
nit: this message is to users, so it should focus less on what the function does, but more on how to fix the problem. Keeping it as simple and as instructive as possible.
Something like this. `Resource(apps.v1/Deployment, exampleNamespace, example App) has unmatched field type in path: spec.tempaltes, expected Slice, got Map`